src/Xamarin.Android.Build.Tasks/Tasks/CreateDynamicFeatureManifest.cs

```diff
@@ -0,0 +1,140 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Xml;
+using System.Xml.Linq;
+using Microsoft.Build.Utilities;
+using Microsoft.Build.Framework;
+
+using Xamarin.Android.Tools;
+using Microsoft.Android.Build.Tasks;
+
+namespace Xamarin.Android.Tasks
+{
+	public class CreateDynamicFeatureManifest : AndroidTask
+	{
+		readonly XNamespace androidNS = "http://schemas.android.com/apk/res/android";
+		readonly XNamespace distNS = "http://schemas.android.com/apk/distribution";
+		readonly XNamespace toolsNS = "http://schemas.android.com/tools";
+
+		public override string TaskPrefix => "CDFM";
+
+		[Required]
+		public string FeatureSplitName { get; set; }
+
+		[Required]
+		public string FeatureDeliveryType { get; set; }
+
+		[Required]
+		public string FeatureType { get; set; }
+
+		[Required]
+		public string PackageName { get; set; }
+
+		[Required]
+		public ITaskItem OutputFile { get; set; }
+
+		public string FeatureTitleResource { get; set; }
+
+		public string MinSdkVersion { get; set; }
+
+		public string TargetSdkVersion { get; set; }
+
+		public bool IsFeatureSplit { get; set; } = false;
+		public bool IsInstant { get; set; } = false;
+		public bool HasCode { get; set; } = false;
+
+		public override bool RunTask ()
+		{
+			XDocument doc = new XDocument ();
+			switch (FeatureType) {
+				case "AssetPack":
+					GenerateAssetPackManifest (doc);
+					break;
+				default:
+					GenerateFeatureManifest (doc);
+					break;
+			}
+			doc.Save (OutputFile.ItemSpec);
```

Comment:
Is there some `XDocument.SaveIfChanged` functionality we could use here? Right now the calling MSBuild target not using `Inputs/Outputs`, that would also be another way.